starter-code/lib/harryPotter2.js

```diff
@@ -0,0 +1,11 @@
+function orderedBirthdays(){
+var nameAndDay = birthdays.filter(function(element,index,array){
+  var newBirthdays=[];
+  if ((index)%2===0){
+    newBirthdays[index]=[element,array[index+1]];
+    }
+return newBirthdays;
+});
+}
+
+console.log(orderedBirthdays());
```

Comment:
Good call using the expression, `index % 2 === 0`. Here is a short explanation why it works. 👍 to verify if this is true or false. true for elements 0,2,4,6 (which are the names), false are elements 1,3,5,7,9 (which are the birthdays)
//Isak

I will give you my solution using reduce and forEach.

```
// Using reduce:
var matrix = birthdays.reduce(function(acc, item, index, array) {
  if (index % 2 === 0) {
    acc.push([array[index], array[index + 1]]);
  }
  return acc;
}, []);

// Using forEach
var nameBirtPairs = [];
birthdays.forEach(function(element, index, array) {
  if (index % 2 === 0) {
    nameBirtPairs.push([array[index], array[index + 1]]);
  }
});
console.log(nameBirtPairs);

// As we do not want any unnecessary variable in the global scope! We should wrap the forEach method inside a function, like this:
function generateNameBirth(array) {
  var nameBirtPairs = [];
  array.forEach(function(element, index, array) {
    if (index % 2 === 0) {
      nameBirtPairs.push([array[index], array[index + 1]]);
    }
  });
  return nameBirtPairs;
}
console.log(generateNameBirth(birthdays));

```